dag.go

```diff
@@ -245,34 +242,22 @@ func createCallback(c *Config, wf WalkCallback) func(v dag.Vertex) (diags dag.Di
 			return diags
 		}
 
-		// if the resource is disabled we need to skip the resource
-		if r.GetDisabled() {
-			return nil
-		}
-
-		// Process the raw resource now we have the context from the linked
-		// resources
-
 		// if there are defaults defined on the resource set them
 		defaults.Set(r)
 
+		// process the raw resource now we have the context from the linked
+		// resources
 		decodeDiags := hcl.Diagnostics{}
 		withContextLock(ctx, func() {
 			decodeDiags = gohcl.DecodeBody(bdy, ctx, r)
 		})
 
 		if decodeDiags.HasErrors() {
-			pe := &errors.ParserError{}
-			pe.Filename = r.Metadata().File
-			pe.Line = r.Metadata().Line
-			pe.Column = r.Metadata().Column
-			pe.Message = fmt.Sprintf(`unable to decode body: %s`, decodeDiags.Error())
 			// this error is set as warning as it is possible that the resource has
 			// interpolation that is not yet resolved
+			parserErr := createParserWarning(r, fmt.Sprintf(`unable to decode body: %s`, decodeDiags.Error()))
```

Comment:
maybe change the wording of the comment above that the error starts off as a warning level, but can be escalated further down after decoding ?

Reply:
done